test/e2e/healthcheck_test.go

```diff
@@ -0,0 +1,189 @@
+//go:build e2e
+
+package e2e
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kuadrant/dns-operator/api/v1alpha1"
+	"github.com/kuadrant/dns-operator/test/e2e/helpers"
+)
+
+const TestTimeoutMedium = 10 * time.Second
+const TestTimeoutLong = 60 * time.Second
+
+// Test Cases covering multiple DNSRecords updating a set of records in a zone
+var _ = Describe("Health Check Test", func() {
+	var dnsRecord1 *v1alpha1.DNSRecord
+	var dnsRecord2 *v1alpha1.DNSRecord
+
+	AfterEach(func(ctx SpecContext) {
+		if dnsRecord1 != nil {
+			err := k8sClient.Delete(ctx, dnsRecord1,
+				client.PropagationPolicy(metav1.DeletePropagationForeground))
+			Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
+		}
+		if dnsRecord2 != nil {
+			err := k8sClient.Delete(ctx, dnsRecord2,
+				client.PropagationPolicy(metav1.DeletePropagationForeground))
+			Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
+		}
+	})
+
+	Context("DNS Provider health checks", func() {
+		It("creates health checks for a health check spec", func(ctx SpecContext) {
+			probeHost := "app." + testZoneDomainName
+			if !slices.Contains(supportedHealthCheckProviders, strings.ToLower(testDNSProvider)) {
+				Skip(fmt.Sprintf("Health checks only supported with DNS Providers: %v", supportedHealthCheckProviders))
+			}
```

Comment:
Should we have some test for providers that do not support health checks so we can test what will happen if you did add a health check?